numpy/ma/core.py

```diff
@@ -4076,11 +4076,10 @@ def __repr__(self):
             # The dtype of the fill value should match that of the array
             # and it is unnecessary to print the full `np.int64(...)`
             # Allow strings: "N/A" has length 3 so would mismatch often.
-            fill_repr = np.core.arrayprint.get_formatter(
-                    dtype=self._fill_value.dtype, fmt=repr)(self.fill_value)
+            fill_repr = str(self.fill_value)
```

Comment:
This doesn't work for strings, we need to do the same thing we do inside structured voids here really, but that is hard now, so I will add a branch for strings and live with the rest of the fallout.